state-transition/core/validation_deposits.go

```diff
@@ -154,27 +154,19 @@ func (sp *StateProcessor[
 			)
 		}
 
-		for i, sd := range localDeposits {
-			// Deposit indices should be contiguous
-			//#nosec:G701 // i never negative
-			expectedIdx := expectedStartIdx + uint64(i)
-			if sd.GetIndex().Unwrap() != expectedIdx {
-				return errors.Wrapf(
-					ErrDepositIndexOutOfOrder,
-					"local deposit index: %d, expected index: %d",
-					sd.GetIndex().Unwrap(), expectedIdx,
-				)
-			}
-
-			if !sd.Equals(deposits[i]) {
-				return errors.Wrapf(
-					ErrDepositMismatch,
-					"local deposit: %+v, payload deposit: %+v",
-					sd, deposits[i],
-				)
-			}
+		// Verify that the local deposits have the same root as the block deposits.
+		var eth1Data *ctypes.Eth1Data
+		eth1Data, err = st.GetEth1Data()
```

Comment:
See my comment above. The `Eth1Data` here was set already to what comes in the block, and it will stay so even if the comparison below fails.